sqlglot/generator.py

```diff
@@ -55,6 +56,7 @@ class Generator:
     """
 
     TRANSFORMS = {
+        **DECODE_TO_CASE,  # type: ignore
```

Comment:
this doesn't seem like the correct approach. if you want all dialects to convert decode, then you should just do it at parse time.

Reply:
It's currently not converted for all dialects, i.e. `DECODE(...)` is transpiled to itself for Snowflake, Redshift, Oracle. Do you think we should just do this at parse time and convert it regardless of the dialect?

Reply:
how is it not converted? is it because they override match to decode?

Reply:
yep

Reply:
this doesn’t follow the sqlglot pattern. the generic dialect should also handle match or decode. 

languages that don’t support decode should use this transform.

Reply:
this way if you do snowflake -> generic -> snowflake it’s still uses decode.

otherwise you lose it